src/tree/updater_gpu_hist.cu

```diff
@@ -511,8 +511,8 @@ struct GPUHistMakerDevice {
     }
 
     CHECK(p_tree);
-    dh::safe_cuda(cudaSetDevice(ctx_->gpu_id));
-    CHECK_EQ(out_preds_d.DeviceIdx(), ctx_->gpu_id);
+    dh::safe_cuda(cudaSetDevice(ctx_->Ordinal()));
+    CHECK(out_preds_d.Device().IsCUDA());
```

Comment:
Shouldn't this line compare the device of `out_preds_d` with `ctx_->Device()`?

Reply:
It's just CUDA, we won't be using GPU hist if context is CPU.

Reply:
What if GPU id mismatches?

Reply:
That's a bug and might segfault due to accessing the wrong memory

Reply:
Yes, so shouldn't we have `CHECK_EQ(out_preds.Device(), ctx_->Device())` ?

Reply:
I will have an additional check for matching context.

Reply:
Done.